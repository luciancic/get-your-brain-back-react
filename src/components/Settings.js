import React, { useContext } from 'react'
import SettingsContext from '../context/SettingsContext'
import SettingsSelect from "./SettingsSelect"

function Settings(props) {
    const { redirectTo } = props
    const [settings, setSettings] = useContext(SettingsContext)
    const options = {
        n: [ 2, 3, 4, 5, 6, 7, 8, 9, 10 ],
        maxRounds: [ 10, 15, 20, 25, 30 ],
        duration: [ 1000, 1500, 2000, 2500, 3000, 3500 ]
    }

    return <div className='container center'>
        <h1>Settings</h1>
        <label>N:</label>
        <SettingsSelect name="n" value={settings.n} options={options.n} changeSetting={(val) => setSettings('n', val)}/>
        <label>Maximum Rounds:</label>
        <SettingsSelect name="maxRounds" value={settings.maxRounds} options={options.maxRounds} changeSetting={(val) => setSettings('maxRounds', val)}/>
        <label>Round Duration</label>
        <SettingsSelect name="duration" value={settings.duration} options={options.duration} changeSetting={(val) => setSettings('duration', val)}/>
        <button className="btn" onClick={ () => redirectTo('Menu') }>Back to Menu</button>
    </div>
}

export default Settings
