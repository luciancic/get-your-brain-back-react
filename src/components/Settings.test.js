import React from 'react';
import { shallow } from 'enzyme';
import { Settings } from './Settings';

it('renders Settings', () => {
  expect( shallow( <Settings /> )).toMatchSnapshot();
});