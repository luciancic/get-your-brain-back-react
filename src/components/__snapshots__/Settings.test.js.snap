// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Settings 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Settings />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Settings
        </h1>,
        <label>
          N:
        </label>,
        <SettingsSelect
          name="n"
        />,
        <label>
          Maximum Rounds:
        </label>,
        <SettingsSelect
          name="maxRounds"
        />,
        <label>
          Round Duration
        </label>,
        <SettingsSelect
          name="duration"
        />,
        <Link
          className="btn"
          replace={false}
          to="/"
        >
          Back to Menu
        </Link>,
      ],
      "className": "container center",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Settings",
        },
        "ref": null,
        "rendered": "Settings",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "N:",
        },
        "ref": null,
        "rendered": "N:",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "name": "n",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Maximum Rounds:",
        },
        "ref": null,
        "rendered": "Maximum Rounds:",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "name": "maxRounds",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Round Duration",
        },
        "ref": null,
        "rendered": "Round Duration",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "name": "duration",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Back to Menu",
          "className": "btn",
          "replace": false,
          "to": "/",
        },
        "ref": null,
        "rendered": "Back to Menu",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Settings
          </h1>,
          <label>
            N:
          </label>,
          <SettingsSelect
            name="n"
          />,
          <label>
            Maximum Rounds:
          </label>,
          <SettingsSelect
            name="maxRounds"
          />,
          <label>
            Round Duration
          </label>,
          <SettingsSelect
            name="duration"
          />,
          <Link
            className="btn"
            replace={false}
            to="/"
          >
            Back to Menu
          </Link>,
        ],
        "className": "container center",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Settings",
          },
          "ref": null,
          "rendered": "Settings",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "N:",
          },
          "ref": null,
          "rendered": "N:",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "name": "n",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Maximum Rounds:",
          },
          "ref": null,
          "rendered": "Maximum Rounds:",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "name": "maxRounds",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Round Duration",
          },
          "ref": null,
          "rendered": "Round Duration",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "name": "duration",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Back to Menu",
            "className": "btn",
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": "Back to Menu",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
